search-relevancy-test/src/search_relevancy_test/edsc_log_parser.clj

```diff
@@ -0,0 +1,123 @@
+(ns search-relevancy-test.edsc-log-parser
+  "Functions for parsing the relevancy logs from EDSC in order to generate anomaly tests."
+  (:require
+   [cheshire.core :as json]
+   [clojure.data.csv :as csv]
+   [clojure.java.io :as io]
+   [clojure.string :as string]))
+
+(def output-filename
+  "edsc_anomaly_tests.csv")
+
+(defn- get-selected-index
+  "Returns the index that was selected in the EDSC log line."
+  [metrics]
+  (Integer/parseInt (:selected_index metrics)))
+
+(defn- get-selected-collection
+  "Returns the collection concept-id that was selected from the EDSC relevancy metric."
+  [metrics]
+  (:selected_collection metrics))
+
+(defn- get-top-collection
+  "Returns the collection that was returned as the first result from the EDSC relevancy metric."
+  [metrics]
+  (first (:collections metrics)))
+
+(defn- get-cmr-search-from-metric
+  "Returns the CMR search that was used from the given EDSC relevancy metric."
+  [metrics]
+  (when-let [free-text (get-in metrics [:query :free_text])]
+    (str "?keyword=" (string/replace (str free-text "*") " " "%20*"))))
+
+(defn get-relevancy-json-from-line
+  "Strips out everything except for the relevancy JSON from an EDSC metrics log line. Return nil
+  if there are any parsing problems"
+  [line]
+  (try
+    (-> (re-find #"\{.*\}" line)
+        (json/parse-string true)
+        :data)
+    (catch Exception e
```

Comment:
I know this is easy to return nil. Don't you at least want to log exceptions so that we can look to find out what the parsing problem was - maybe fix the EDSC metrics log?  Just a suggestion because if you have 1 error you are probably going to have many.

Reply:
I logged the exception at warning level.